File: VB/Controllers/HomeController.vb
Imports System
Imports System.IO
Imports System.Web
Imports System.Web.Mvc
Imports DevExpress.XtraPrinting
Imports DevExpress.XtraPrintingLinks
Imports DevExpress.Web.Mvc
Imports CS.Model
Imports DevExpress.Web

Namespace CS.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View(New MyViewModel With {.Products = MyModel.GetProducts()})
        End Function

        Public Function GridViewPartialProducts() As ActionResult
            Return PartialView(MyModel.GetProducts())
        End Function

        Public Function ExportTo() As ActionResult
            Dim exportOptions As New XlsxExportOptionsEx()
            AddHandler exportOptions.CustomizeCell, AddressOf exportOptions_CustomizeCell
            Return GridViewExtension.ExportToXlsx(GridViewHelper.ExportGridViewSettings, MyModel.GetProducts(), exportOptions)
        End Function

        Private Sub exportOptions_CustomizeCell(ByVal ea As DevExpress.Export.CustomizeCellEventArgs)
            If ea.ColumnFieldName = "UnitPrice" Then
                If Convert.ToInt32(ea.Value) > 15 Then
                    ea.Formatting.BackColor = System.Drawing.Color.Yellow
                Else
                    ea.Formatting.BackColor = System.Drawing.Color.Green
                End If
                ea.Handled = True
            End If

        End Sub

    End Class

End Namespace
Public NotInheritable Class GridViewHelper

    Private Sub New()
    End Sub

    Private Shared exportGridViewSettings_Renamed As GridViewSettings

    Public Shared ReadOnly Property ExportGridViewSettings() As GridViewSettings
        Get
            If exportGridViewSettings_Renamed Is Nothing Then
                exportGridViewSettings_Renamed = CreateExportGridViewSettings()
            End If
            Return exportGridViewSettings_Renamed
        End Get
    End Property

    Private Shared Function CreateExportGridViewSettings() As GridViewSettings
        Dim settings As New GridViewSettings()

        settings.Name = "gvProducts"
        settings.CallbackRouteValues = New With {Key .Controller = "Home", Key .Action = "GridViewPartialProducts"}

        settings.KeyFieldName = "ProductID"
        settings.Settings.ShowFilterRow = True

        settings.Columns.Add("ProductID")
        settings.Columns.Add("ProductName")
        settings.Columns.Add("UnitPrice")

        Return settings
    End Function
End Class

File: CS/Controllers/HomeController.cs
using System;
using System.IO;
using System.Web;
using System.Web.Mvc;
using DevExpress.XtraPrinting;
using DevExpress.XtraPrintingLinks;
using DevExpress.Web.Mvc;
using CS.Model;
using DevExpress.Web;
using System.Globalization;

namespace CS.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View(new MyViewModel { Products = MyModel.GetProducts() });
        }

        public ActionResult GridViewPartialProducts() {
            return PartialView(MyModel.GetProducts());
        }

        public ActionResult ExportTo() {
            XlsxExportOptionsEx exportOptions = new XlsxExportOptionsEx();
            exportOptions.CustomizeCell += new DevExpress.Export.CustomizeCellEventHandler(exportOptions_CustomizeCell);
            return GridViewExtension.ExportToXlsx(GridViewHelper.ExportGridViewSettings, MyModel.GetProducts(), exportOptions);
        }

        void exportOptions_CustomizeCell(DevExpress.Export.CustomizeCellEventArgs ea) {
            if(ea.AreaType != DevExpress.Export.SheetAreaType.Header && ea.ColumnFieldName == "UnitPrice") {
                if(Convert.ToDecimal(ea.Value) > 15)
                    ea.Formatting.BackColor = System.Drawing.Color.Yellow;
                else
                    ea.Formatting.BackColor = System.Drawing.Color.Green;
                ea.Handled = true;
            }
        }
    }

}
public static class GridViewHelper {
    private static GridViewSettings exportGridViewSettings;

    public static GridViewSettings ExportGridViewSettings {
        get {
            if(exportGridViewSettings == null)
                exportGridViewSettings = CreateExportGridViewSettings();
            return exportGridViewSettings;
        }
    }

    private static GridViewSettings CreateExportGridViewSettings() {
        GridViewSettings settings = new GridViewSettings();

        settings.Name = "gvProducts";
        settings.CallbackRouteValues = new { Controller = "Home", Action = "GridViewPartialProducts" };

        settings.KeyFieldName = "ProductID";
        settings.Settings.ShowFilterRow = true;

        settings.Columns.Add("ProductID");
        settings.Columns.Add("ProductName");
        settings.Columns.Add("UnitPrice");

        return settings;
    }
}
